src/Router/RouteCollectorDelegator.php

```diff
@@ -29,7 +30,7 @@ public function __invoke(
         Assert::isInstanceOf($collector, RouteCollectorInterface::class);
 
         $config = $container->get('config');
-        Assert::isArray($config);
+        assert(is_array($config) || $config instanceof ArrayAccess);
```

Comment:
Where assert library is available check would be
```suggestion
        Assert::isArrayAccessible($config);
```